cai_causal_graph/time_series_causal_graph.py

```diff
@@ -1007,6 +1007,50 @@ def to_numpy_by_lag(self) -> Tuple[Dict[int, numpy.ndarray], List[str]]:
         assert self.variables is not None
         return adjacency_matrices, self.variables
 
+    def get_nodes_at_lag(self, time_lag: int = 0) -> List[TimeSeriesNode]:
+        """Return all nodes at time delta `time_lag`."""
+        return [node for node in self.get_nodes() if node.time_lag == time_lag]
+
+    def get_contemporaneous_nodes(self, node: NodeLike) -> List[TimeSeriesNode]:
+        """Return all nodes that are contemporaneous to the provided node."""
+        assert node is not None, 'The `node` cannot be None.'
+        if isinstance(node, str):
+            node = self.get_node(node)
+
+        contemporaneous_nodes = []
+        assert isinstance(node, TimeSeriesNode), 'The node must be a `TimeSeriesNode`.'
+
+        for other_node in self.get_nodes():
+            assert isinstance(other_node, TimeSeriesNode)
+            if other_node.identifier != node.identifier and other_node.time_lag == node.time_lag:
+                contemporaneous_nodes.append(node)
+        return contemporaneous_nodes
+
+    def get_contemporaneous_adj_nodes(self, node: NodeLike) -> List[TimeSeriesNode]:
+        """Return the adjacent that are contemporanous to the provided node."""
+        assert node is not None, 'The `node` cannot be None.'
+        if isinstance(node, str):
+            node = self.get_node(node)
+
+        assert isinstance(node, TimeSeriesNode), 'The node must be a `TimeSeriesNode`.'
+
+        contemporaneous_adj_nodes = []
+
+        # go through the inbounds edges
+        for edge in self.get_edges(destination=node):
+            assert isinstance(edge.source, TimeSeriesNode)
+            if edge.source.time_lag == node.time_lag:
```

Comment:
According to your changelog, this should only add edges if they're directed, but you don't check that?

Reply:
I want both directed and undirected. I just want all the nodes that may be parents. Updated the changelog.